clang/lib/Parse/ParseDecl.cpp

```diff
@@ -2933,6 +2933,9 @@ Decl *Parser::ParseDeclarationAfterDeclaratorAndAttributes(
   }
   case InitKind::Uninitialized: {
     Actions.ActOnUninitializedDecl(ThisDecl);
+    if (D.isDecompositionDeclarator())
+      Diag(Tok, diag::err_decomp_decl_expected_init)
+          << dyn_cast<VarDecl>(ThisDecl) << PP.getSpelling(Tok);
```

Comment:
I think this `dyn_cast<>` should be a `cast<>` instead.

Also, I think this code should continue to live in `ActOnUninitializedDecl` so anyone calling the function gets the benefit of the diagnostic. Does using `Var->getEndLoc()` get the caret into the correct location?